extensions/agent/src/mainController.ts

```diff
@@ -169,7 +173,7 @@ export class MainController {
 			if (!ownerUri || ownerUri instanceof vscode.Uri) {
 				let path: string;
 				if (!ownerUri) {
-					if (azdata.nb.activeNotebookEditor.document.isDirty) {
+					if (azdata.nb.activeNotebookEditor.document.isDirty || azdata.nb.activeNotebookEditor.document.isUntitled) {
 						vscode.window.showErrorMessage(localize('agent.unsavedFileSchedulingError', 'Save file before scheduling'), { modal: true });
```

Comment:
While you're here - this message isn't very friendly. I'd suggest updating it to

`The notebook must be saved before being scheduled. Please save and then retry scheduling again.`